lib/rubygems/request.rb

```diff
@@ -237,6 +237,14 @@ def perform_request(request) # :nodoc:
       verbose "fatal error"
 
       raise Gem::RemoteFetcher::FetchError.new('fatal error', @uri)
+    rescue (defined?(OpenSSL::SSL) ? OpenSSL::SSL::SSLError : nil) => e
+      # As of 2018-11, error is intermittently raised when using TLS 1.3 with
+      # improper client authentication
+      if OpenSSL::SSL.const_defined? :TLS1_3_VERSION
```

Comment:
Should this be checked earlier in https://github.com/rubygems/rubygems/blob/fccc3139d4bf0924af090e487b3e752697fe8e4c/lib/rubygems/request.rb#L134 since we already checking if `OpenSSL::SSL` is defined there?

Reply:
I believe all of the caller listings show hitting RG first at request.rb:222.  The rescue in 240 catches errors in line 222.